react/components/EXPERIMENTAL_TableTree/Tree/TreeHeadings.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { FC, useContext } from 'react'
+import uuid from 'uuid'
+
+import { TABLE_HEADER_HEIGHT } from '../../EXPERIMENTAL_Table/constants'
+import { useTableContext } from '../../EXPERIMENTAL_Table/contexts'
+import { useCheckboxesContext } from '../contexts'
+import { Row, CellProps, RowProps } from '../../EXPERIMENTAL_Table/Styled'
+import CellPrefix from './CellPrefix'
+
+const TreeHeadings: FC<TreeHeadingsProps> = ({ cellProps, rowProps }) => {
+  const { visibleColumns } = useTableContext()
+  const {
+    toggle,
+    itemTree,
+    isChecked,
+    isPartiallyChecked,
+  } = useCheckboxesContext()
+
+  return (
+    <Row {...rowProps} height={TABLE_HEADER_HEIGHT}>
+      {visibleColumns.map((headerData: Column, headerIndex: number) => {
+        const { headerRender, title, width } = headerData
+        const content = headerRender ? headerRender({ headerData }) : title
+        return (
+          <Row.Cell
+            {...cellProps}
+            className="bt"
+            key={`heading-${uuid()}`}
```

Comment:
This uuid is great, didn't wonder something like that existed

Reply:
Yeah! It is awesome